Add binwarp mouse control mode

aartaka opened the pull request:
This add a new interactive-keymap-based mouse control mode (in the sense of `iresize` mode and maybe even Emacs modes) -- Binwarp. It operates on restricted screen area and divides it in two on every action, making mouse movement into a recursive binary divide-and-conquer task.

Also, the `beacon` README entry was added when I ran `update-readme.sh`, so I included it in the PR as a separate commit.

### Motivation

This mode appeared when my laptop touchpad died and I felt a need in going mouse-less in domain of mouse control :) For now, this module is already covering all my needs and even has some configuration bits that I don't use, so I thought it can be useful for someone else.

If this is more suitable for `minor-mode` directory, I'll move it there.

I'll be glad if you could review it and point at some possibly unconventional or optimizable things!

PuercoPop commented on util/binwarp/binwarp.lisp:
```diff
@@ -0,0 +1,105 @@
+
+(in-package #:binwarp)
+
+(declaim (special *binwarp-mode-p* *binwarp-area* *binwarp-history*
+                  *reinitiate-ptr* *init-ptr-position*
+                  *preserve-history*))
```
```suggestion
```

You do no't need to declaim the variables as special for variables that are defined in the same file.

PuercoPop commented on util/binwarp/binwarp.lisp:
```diff
@@ -0,0 +1,105 @@
+                  *preserve-history*))
+(defvar *binwarp-mode-p*     nil "Whether you are in the binwarp mode.")
+(defvar *binwarp-area*       nil "The current screen area that binwarp is restricted by.")
+(defvar *binwarp-history*    nil "The previous binwarp areas.")
```
If this is a list, consider hinting so by using `()`, which is a synonym of `nil`.

```suggestion
(defvar *binwarp-history*    () "The previous binwarp areas.")
```

aartaka replied:
Shouldn't it be a literal list (`'()`) ? Or is empty lists quoting optional?

PuercoPop replied:
In CL we don't need to quote `()`. Only in Scheme. Some people recommend doing so anyway and although I disagree that is the style that you'll find in the StumpWM codebase.

PuercoPop commented on util/binwarp/user-facing.lisp:
```diff
@@ -0,0 +1,97 @@
+
+(in-package :stumpwm)
+
+;;; This file exists because I needed a good and obvious way
+;;; for user to define their own binwarp-mode settings while being in the
+;;; :STUMPWM package. If there's a better way, I'll be glad to hear about it!
```
Before I can suggest a better™ way I need to understand why you need this to be in the stumpwm package.

aartaka replied:
Sure. The reason for that was that the symbol for `kbd` is not present in `binwarp` package and, when I was uquote-splicing (`,@`) the list of keybindings, `kbd` was automatically interned to `binwarp`. Because it was unbing in `binwarp`, it was causing errors during macroexpansion.

The only solution I came up with was to `defvar` the `*default-binwarp-keymap*` in the `:stumpwm` package and refer to it in `define-binwarp-mode` in the same package.

So, if there is a way to make `kbd` in the `*default-binwarp-keymap*` to be interpreted as `stumpwm:kbd` and not `binwarp:kbd`, then I won't need to pollute `:stumpwm`.

aartaka commented:
> There is only issue is that the README should be an org file instead of a markdown file. The update-readme.sh hardcodes that assumption. You can see the links in the top-level README pointing to `./util/beckon/README.org` and `./util/binwrap/README.org`.

Oh, I haven't noticed that, sorry :) I'll be happy to rewrite it as org file!

aartaka commented:
I've addressed your suggestions:
- `*binwarp-history*` is now explicitly defined as list.
- unnecessary `special` declarations are gone.
- `user-facing.lisp` file is deleted and utils from it are moved to `binwarp.lisp`, because my problem with problematic interning was solved by adding `stumpwm:` prefix to `kbd`.
